public/js/main.js

```diff
@@ -705,6 +726,7 @@ async function main() {
     try {
         validateGlobalQueryParams(queryParams.global);
         query = JSON.parse((await getQuery()).response).body;
+        validateQueryId(query);
```

Comment:
Why this call is needed here and what's the overall need of this extra function?

Looking at the code, `queryParams.global.queryId` is only used in `getQuery()` called at the line before. How this line is going to do anything significant for the rest of the script execution?

Reply:
`queryParams` is passed to the backend for running the query (line 101): the backend uses the queryId in the `queryParams` rather then the one contained in the `query` object: changing this behavior in the backend would also have required a number of other changes, and to limit the changes to the backend logic I preferred to add the control in the frontend.

Basically, this validation step creates `queryParams.global.queryId` if it's missing by taking the queryId from the retrieved query